common/src/main/java/com/wynntils/features/ui/PlayerInfoFeature.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.ui;
+
+import com.mojang.blaze3d.platform.Window;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.config.Category;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigCategory;
+import com.wynntils.core.config.ConfigHolder;
+import com.wynntils.core.config.RegisterConfig;
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.features.overlays.Overlay;
+import com.wynntils.core.features.overlays.OverlayPosition;
+import com.wynntils.core.features.overlays.annotations.OverlayInfo;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.RenderEvent;
+import com.wynntils.utils.StringUtils;
+import com.wynntils.utils.colors.CustomColor;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.Texture;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.components.PlayerTabOverlay;
+import net.minecraft.client.multiplayer.PlayerInfo;
+import net.minecraft.client.renderer.MultiBufferSource;
+import org.joml.Math;
+
+@ConfigCategory(Category.OVERLAYS)
+public class PlayerInfoFeature extends Feature {
+    private static final Comparator<PlayerInfo> PLAYER_INFO_COMPARATOR =
+            Comparator.comparing(playerInfo -> playerInfo.getProfile().getName(), String::compareToIgnoreCase);
+    private static final int DISTANCE_BETWEEN_CATEGORIES = 87;
+    private static final int ROLL_WIDTH = 27;
+    private static final int HALF_WIDTH = 178;
+    private static final int WIDTH = HALF_WIDTH * 2;
+    private static final int TOTAL_WIDTH = WIDTH + ROLL_WIDTH * 2;
+    private static final int MAX_WIDTH = 73;
+
+    @OverlayInfo(renderType = RenderEvent.ElementType.GUI)
+    public final PlayerInfoOverlay playerInfoOverlay = new PlayerInfoOverlay();
+
+    private static class PlayerInfoOverlay extends Overlay {
```

Comment:
I am not even sure if this is an actual overlay and if it should be one. What do you think @magicus?

Reply:
Maybe...? I agree that it is a bit borderline. I have not studied the code fully, but if we indeed do the entire rendering of the tab list, and not just draw a pretty background, then I think it can and should be an overlay. Since you can then move it around the screen, etc. That it is not showing anything most of the time is not really a problem, we have other overlays that behave like that, like the NPC dialogue overlay.

If we just decorate the existing vanilla tab list, so things will look messed up if we move this overlay around, then it should definitely not be an overlay.